### PR TITLE
Add CloudWatch input plugin

### PR DESCRIPTION
**Pull request by joshhardy**
@Glympse/developers Please take a look

### CODE REVIEW

File: plugins/inputs/cloudwatch/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# Amazon CloudWatch Statistics Input
+
+This plugin will pull Metric Statistics from Amazon CloudWatch.
+
+### Amazon Authentication
+
+This plugin uses a credential chain for Authentication with the CloudWatch
+API endpoint. In the following order the plugin will attempt to authenticate.
+1. [IAMS Role](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html)
+2. [Environment Variables](https://github.com/aws/aws-sdk-go/wiki/configuring-sdk)
```
Comment (schuyler-glympse) [inline plugins/inputs/cloudwatch/README.md]: Nitpick: https://github.com/aws/aws-sdk-go/wiki/configuring-sdk#environment-variables

File: plugins/inputs/cloudwatch/README.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+3. [Shared Credentials](https://github.com/aws/aws-sdk-go/wiki/configuring-sdk)
```
Comment (schuyler-glympse) [inline plugins/inputs/cloudwatch/README.md]: Nitpick: https://github.com/aws/aws-sdk-go/wiki/configuring-sdk#shared-credentials-file

**Comment by schuyler-glympse** [comment]
@joshglympse This LGTM, but since my Go knowledge is still around beginner level, let's make sure someone else says it is g2g.

**Comment by nullpuppy** [comment]
LGTM

**Pull request merged**